pkg/visor/logserver/api.go

```diff
@@ -32,114 +27,200 @@ type API struct {
+	// Gin router without default logging
+	r := gin.New()
+	// use Gin's recovery logging middleware to recover from panic
+	r.Use(gin.Recovery())
 	if printLog {
-		r.Use(middleware.Logger)
-		r.Use(middleware.Recoverer)
+		// use custom logging middleware
+		r.Use(loggingMiddleware())
 	}
-	r.Use(httputil.SetLoggerMiddleware(log))
-
-	r.Get("/health", api.health)
 
-	fsTP := http.FileServer(http.Dir(tpLogPath))
-	r.Handle("/*", http.StripPrefix("/", fsTP))
-
-	fsLocal := http.FileServer(http.Dir(localPath))
-	r.Handle("/"+skyenv.NodeInfo, http.StripPrefix("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		whitelistFSHandler(w, r, localPath, whitelistedPKs)
-	})))
+	// whitelist-based authentication for survey collection if there are keys whitelisted for that
+	// no survey-whitelisted keys means the file is publicly accessible
+	authRoute := r.Group("/")
+	if len(whitelistedPKs) > 0 {
+		authRoute.Use(whitelistAuth(whitelistedPKs))
+	}
+	// note that the survey only exists / is generated if the reward address is set
+	authRoute.StaticFile("/node-info.json", filepath.Join(localPath, "node-info.json"))
 
-	r.Handle("/transport_logs/*", http.StripPrefix("/", fsLocal))
+	r.GET("/health", func(c *gin.Context) {
+		api.health(c)
+	})
 
-	fsCustom := http.FileServer(http.Dir(customPath))
-	r.Handle("/*", http.StripPrefix("/", fsCustom))
+	// serve transport log files ; then any files in the custom path
+	r.GET("/:file", func(c *gin.Context) {
+		// files with .csv extension are likely transport log files
+		if filepath.Ext(c.Param("file")) == ".csv" {
+			// check transport logs dir for the file, and serve it if it exists
+			_, err := os.Stat(filepath.Join(tpLogPath, c.Param("file")))
+			if err == nil {
+				c.File(filepath.Join(tpLogPath, c.Param("file")))
+				return
+			}
+		}
+		// Check for any file in custom dmsghttp dir path
+		_, err := os.Stat(filepath.Join(customPath, c.Param("file")))
+		if err == nil {
+			c.File(filepath.Join(customPath, c.Param("file")))
+			return
+		}
+		// File not found, return 404
+		c.Writer.WriteHeader(http.StatusNotFound)
+	})
 
 	api.Handler = r
 	return api
 }
 
-func (api *API) health(w http.ResponseWriter, r *http.Request) {
-	info := buildinfo.Get()
-	api.writeJSON(w, r, http.StatusOK, httputil.HealthCheckResponse{
-		BuildInfo: info,
-		StartedAt: api.startedAt,
-	})
-}
-
-func (api *API) writeJSON(w http.ResponseWriter, r *http.Request, code int, object interface{}) {
-	jsonObject, err := json.Marshal(object)
+func (api *API) health(c *gin.Context) {
+	jsonObject, err := json.Marshal(
+		httputil.HealthCheckResponse{
+			BuildInfo: buildinfo.Get(),
+			StartedAt: api.startedAt,
+		})
 	if err != nil {
-		api.log(r).WithError(err).Errorf("failed to encode json response")
-		w.WriteHeader(http.StatusInternalServerError)
-
+		httputil.GetLogger(c.Request).WithError(err).Errorf("failed to encode json response")
+		c.Writer.WriteHeader(http.StatusInternalServerError)
```

Comment:
Is it necessary to explicitly write the headers if we return errorcode anyways on the next line?

Reply:
we don't actually need to write the error code outside of the header, but currently dmsgget is currently ignorant of the error codes. So in the instance of failure, you can't tell from the client basically.

For my specific purposes of the reward system, i need that logging on a temporary basis.

after dmsgget is fixed, then it should be removed